src/System.Net.Security/src/System/Net/Security/SslState.cs

```diff
@@ -1303,39 +1303,6 @@ internal Task CheckEnqueueWriteAsync()
             }
         }
 
-        internal void CheckEnqueueWrite()
-        {
-            // Clear previous request.
-            _queuedWriteStateRequest = null;
-            int lockState = Interlocked.CompareExchange(ref _lockWriteState, LockWrite, LockNone);
-            if (lockState != LockHandshake)
-            {
-                // Proceed with write.
-                return;
-            }
```

Comment:
It looks like we don't have any tests that make it beyond this return, in either the sync or async versions.  Is it possible to add some?

Reply:
In a word no, @bartonjs should be able to confirm. But from .net I am sure there is no way to trigger a renegotiation. The testing I have done has been pretty manual using OpenSSL, or my own code to trigger the situations. So it might be possible to write an outerloop style scripted test but that would be beyond my knowledge of how your environment works for outerloop stuff...

Reply:
Ok, thanks.  It's good to hear you've tested it manually, thanks, but it's scary to me that we have all this code to support renegotiation and nothing actually testing it automatically.  @bartonjs, @Priya91, @geoffkizer, @wfurt, anything we can do about that?

Reply:
Yes I agree, if someone is happy to help me through how a scripted outloop test is created, I am more than happy to put something together that should test it, if help is wanted. (I also agree that its a bit worrying that there is zero test coverage for what is a complex bit of code).

https://github.com/dotnet/corefx/pull/22998#issuecomment-320800860